### PR TITLE
Add the following flags to buildah bud and from

### PR DESCRIPTION
**Pull request by umohnani8**
Add the following flags to buildah bud and from

        --add-host
        --cgroup-parent
        --cpu-period
        --cpu-quota
        --cpu-shares
        --cpuset-cpus
        --cpuset-mems
        --memory
        --memory-swap
        --security-opt
        --ulimit

Signed-off-by: umohnani8 <umohnani@redhat.com>

### CODE REVIEW

**Comment by umohnani8** [comment]
@rhatdan @nalind PTAL

**Comment by rhatdan** [comment]
Tests don't like your complexity.

**Comment by TomSweeneyRedHat** [comment]
bot, retest this please

**Comment by TomSweeneyRedHat** [comment]
bot looks to have ignored me.  Going to close and reopen to retest.

**Comment by umohnani8** [comment]
Tests are finally green! @rhatdan @TomSweeneyRedHat @nalind PTAL

**Comment by rhatdan** [comment]
Nit on your spelling mistake in check in messages folowing.
 Add the folowing flags to buildah bud and from

File: contrib/completions/bash/buildah [hunk 1]
```diff
@@ -352,20 +352,31 @@ return 1
...
+     -m
+     --memory
+     --memory-swap
      --runtime
      --runtime-flag
      --security-opt
      --signature-policy
      -t
      --tag
+     --ulimt
```
Comment (rhatdan) [inline contrib/completions/bash/buildah]: ulimit

File: docs/buildah-bud.md [hunk 2]
```diff
@@ -82,6 +177,23 @@ runtime, the manpage to consult is `runc(8)`).
...
+**--security-opt**=[]
+
+Security Options
+
+  "label=user:USER"   : Set the label user for the container
+  "label=role:ROLE"   : Set the label role for the container
+  "label=type:TYPE"   : Set the label type for the container
+  "label=level:LEVEL" : Set the label level for the container
+  "label=disable"     : Turn off label confinement for the container
+  "no-new-privileges" : Disable container processes from gaining additional privileges
```
Comment (rhatdan) [inline docs/buildah-bud.md]: Not supported?

File: docs/buildah-from.md [hunk 3]
```diff
@@ -70,6 +164,23 @@ Pull the image even if a version of the image is already present.
...
+**--security-opt**=[]
+
+Security Options
+
+  "label=user:USER"   : Set the label user for the container
+  "label=role:ROLE"   : Set the label role for the container
+  "label=type:TYPE"   : Set the label type for the container
+  "label=level:LEVEL" : Set the label level for the container
+  "label=disable"     : Turn off label confinement for the container
+  "no-new-privileges" : Disable container processes from gaining additional privileges
```
Comment (rhatdan) [inline docs/buildah-from.md]: Not supported.

**Comment by rhatdan** [comment]
Can you add some tests?  Stuff we can steel from Podman?

File: contrib/completions/bash/buildah (hunk 1 above)
Comment (umohnani8) [reply contrib/completions/bash/buildah]: fixed

File: docs/buildah-bud.md (hunk 2 above)
Comment (umohnani8) [reply docs/buildah-bud.md]: fixed

File: docs/buildah-from.md (hunk 3 above)
Comment (umohnani8) [reply docs/buildah-from.md]: fixed

**Comment by umohnani8** [comment]
Added tests and fixed the typo in the commit message.

File: tests/from.bats [hunk 4]
```diff
@@ -128,3 +128,57 @@ load helpers
   buildah rm ${cid}
   buildah rmi alpine alpine2
 }
+
+@test "from cpu-period test" {
+	cid=$(buildah from --cpu-period=5000 --pull --signature-policy ${TESTSDIR}/policy.json alpine)
+  run buildah run $cid cat /sys/fs/cgroup/cpu/cpu.cfs_period_us
```
Comment (rhatdan) [inline tests/from.bats]: Nit can you fix the tabs/indent here?

File: buildah.go [hunk 5]
```diff
@@ -136,6 +137,34 @@ func GetBuildInfo(b *Builder) BuilderInfo {
 	}
 }
 
+// CommonBuildOptions are reseources that can be defined by flags for both buildah from and bud
```
Comment (TomSweeneyRedHat) [inline buildah.go]: FWIW, a nice approach, I like it.  Much better than dumping them all into BuilderOptions individually.

File: cmd/buildah/common.go [hunk 6]
```diff
@@ -217,3 +226,161 @@ func validateFlags(c *cli.Context, flags []cli.Flag) error {
...
+		Name:  "security-opt",
+		Usage: "security Options (default [])",
+	},
+	cli.StringSliceFlag{
+		Name:  "ulimit",
+		Usage: "ulimit options (default [])",
+	},
+}
+
+func parseRunAndBudOpts(c *cli.Context) (*buildah.CommonBuildOptions, error) {
```
Comment (TomSweeneyRedHat) [inline cmd/buildah/common.go]: should this be parseFromAndBudOpts()?

File: docs/buildah-bud.md [hunk 7]
```diff
@@ -82,6 +177,23 @@ runtime, the manpage to consult is `runc(8)`).
 Note: Do not pass the leading `--` to the flag. To pass the runc flag `--log-format json`
 to buildah bud, the option given would be `--runtime-flag log-format=json`.
 
+**--security-opt**=[]
+
+Security Options
```
Comment (TomSweeneyRedHat) [inline docs/buildah-bud.md]: It would be good to have a sample of this one.

File: docs/buildah-bud.md [hunk 8]
```diff
@@ -30,6 +37,73 @@ resulting image's configuration.
 Use certificates at *path* (*.crt, *.cert, *.key) to connect to the registry.
 Default certificates directory is _/etc/containers/certs.d_.
 
+**--cgroup-parent**=""
```
Comment (TomSweeneyRedHat) [inline docs/buildah-bud.md]: An example for this one would be good too.

File: docs/buildah-from.md [hunk 9]
```diff
@@ -80,6 +191,10 @@ option be used, as the default behavior of using the system-wide default policy
 
 Require HTTPS and verify certificates when talking to container registries (defaults to true)
 
+**--ulimit**=[]
```
Comment (TomSweeneyRedHat) [inline docs/buildah-from.md]: An example of this would be good

File: cmd/buildah/common.go (hunk 6 above)
Comment (TomSweeneyRedHat) [reply cmd/buildah/common.go]: Or better yet, parseCommonBuildOptions().

**Comment by TomSweeneyRedHat** [comment]
Oh, forgot to mention, you'll need to add these to libpod build too, but that will be straight forward.

File: cmd/buildah/common.go (hunk 6 above)
Comment (umohnani8) [reply cmd/buildah/common.go]: oh right, it was supposed to be for run first, forgot to change the name. Thanks!

**Comment by umohnani8** [comment]
> I'm fine with this going in now. A few nits here and there. I didn't see any tests for buds.bats and I don't think you'd tests for all of the new common options. It would be nice to get those added in a follow up PR or commit if possible.

@TomSweeneyRedHat it is hard to test these options for buildah bud because the values are not retained when the container is created using buildah form. These options are to limit the resources when the build itself is happening. It is something QE can test through debug statements though.

**Comment by TomSweeneyRedHat** [comment]
Ok on the buildah bud tests then.  I'm sure QE will be looking forward to some detailed qe tips then.

File: tests/from.bats (hunk 4 above)
Comment (umohnani8) [reply tests/from.bats]: fixed

File: cmd/buildah/common.go (hunk 6 above)
Comment (umohnani8) [reply cmd/buildah/common.go]: fixed

File: docs/buildah-bud.md (hunk 7 above)
Comment (umohnani8) [reply docs/buildah-bud.md]: fixed

File: docs/buildah-bud.md (hunk 8 above)
Comment (umohnani8) [reply docs/buildah-bud.md]: fixed

File: docs/buildah-from.md (hunk 9 above)
Comment (umohnani8) [reply docs/buildah-from.md]: fixed

**Comment by rhatdan** [comment]
@rh-atomic-bot r+

**Comment by rhatdan** [comment]
@umohnani8 I think there were a few more docker build options, that we currently don't implement, could you look into those.

**Pull request closed without merging**